File: C#/0020. Valid Parentheses.cs
public class Solution {
    public bool IsValid(string s) {
        Stack<char> sStack=new Stack<char>();
        foreach(char ss in s){
            if (sStack.Count==0){
                sStack.Push(ss);
            }
            else{
                if (ss==')' && sStack.Peek()=='('){
                    sStack.Pop();
                }
                else if(ss==']' && sStack.Peek()=='['){
                    sStack.Pop();
                }else if(ss=='}' && sStack.Peek()=='{'){
                    sStack.Pop();
                }
                else{
                    sStack.Push(ss);
                }
            }
        }
        return sStack.Count==0;
    }
}

File: 20200215Python-China/0020. Valid Parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        s_stack=[]
        for ss in s:
            if ss in "{([":
                s_stack.append(ss)
            else:
                if not s_stack:
                    return False
                if ss==']' and s_stack[-1]=="[":
                    s_stack.pop()
                elif ss==']' and s_stack[-1]=="[":
                    s_stack.pop()
                elif ss=='}' and s_stack[-1]=="{":
                    s_stack.pop()
                elif ss==')' and s_stack[-1]=="(":
                    s_stack.pop()
                else:
                    return False
        return len(s_stack)==0
